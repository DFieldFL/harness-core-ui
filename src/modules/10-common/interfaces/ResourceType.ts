export enum ResourceType {
  PROJECT = 'Project',
  SECRET = 'Secret',
  ORGANIZATION = 'Organization',
  CONNECTOR = 'Connector',
  SERVICE = 'service',
  PIPELINE = 'Pipeline'
}
