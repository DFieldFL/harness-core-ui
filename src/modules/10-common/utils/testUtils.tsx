import React from 'react'
import { UseGetProps, UseGetReturn, RestfulProvider } from 'restful-react'
import { compile } from 'path-to-regexp'
import { createMemoryHistory } from 'history'
import { Router, Route, Switch, useLocation } from 'react-router-dom'
import { ModalProvider } from '@wings-software/uikit'
import qs from 'qs'

import strings from 'strings/strings.en.yaml'
import { AppStoreProvider, useAppStoreWriter } from 'framework/hooks/useAppStore'
import type { AppStore } from 'framework/types/AppStore'
import { AUTH_ROUTE_PATH_PREFIX, StringsContext } from 'framework/exports'

export type UseGetMockData<TData, TError = undefined, TQueryParams = undefined, TPathParams = undefined> = Required<
  UseGetProps<TData, TError, TQueryParams, TPathParams>
>['mock']

export interface UseMutateMockData<TData, TRequestBody = unknown> {
  loading?: boolean
  mutate?: (data?: TRequestBody) => Promise<TData>
}

export type UseGetReturnData<TData, TError = undefined, TQueryParams = undefined, TPathParams = undefined> = Omit<
  UseGetReturn<TData, TError, TQueryParams, TPathParams>,
  'absolutePath' | 'cancel' | 'response'
>

export const findDialogContainer = (): Element | null => document.querySelector('.bp3-dialog')
export const findPopoverContainer = (): Element | null => document.querySelector('.bp3-popover-content')

interface TestWrapperProps {
  path?: string
  pathParams?: Record<string, string | number>
  queryParams?: Record<string, any>
  defaultAppStoreValues?: Partial<AppStore>
}

export const prependAccountPath = (path: string): string => AUTH_ROUTE_PATH_PREFIX + path

const AppStoreInitializer: React.FC<{ defaultAppStoreValues: TestWrapperProps['defaultAppStoreValues'] }> = ({
  children,
  defaultAppStoreValues
}) => {
  const updateAppStore = useAppStoreWriter()
  updateAppStore(defaultAppStoreValues || {})
  return <> {children} </>
}

export const NotFound = (): JSX.Element => {
  const location = useLocation()
  return (
    <div>
      <h1>Not Found</h1>
      <div data-testid="location">{`${location.pathname}${
        location.search ? `?${location.search.replace(/^\?/g, '')}` : ''
      }`}</div>
    </div>
  )
}

export const TestWrapper: React.FC<TestWrapperProps> = props => {
  const { path = '/', pathParams = {}, defaultAppStoreValues, queryParams = {} } = props

  const search = qs.stringify(queryParams, { addQueryPrefix: true })
  const routePath = compile(path)(pathParams) + search

  const history = createMemoryHistory({ initialEntries: [routePath] })

  return (
    <AppStoreProvider>
      <StringsContext.Provider value={strings}>
        <AppStoreInitializer defaultAppStoreValues={defaultAppStoreValues}>
          <Router history={history}>
            <ModalProvider>
              <RestfulProvider base="/">
                <Switch>
                  <Route exact path={path}>
                    {props.children}
                  </Route>
                  <Route>
                    <NotFound />
                  </Route>
                </Switch>
              </RestfulProvider>
            </ModalProvider>
          </Router>
        </AppStoreInitializer>
      </StringsContext.Provider>
    </AppStoreProvider>
  )
}
