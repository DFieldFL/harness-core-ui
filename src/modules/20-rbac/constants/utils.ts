export enum RbacResourceGroupTypes {
  DYNAMIC_RESOURCE_SELECTOR = 'DynamicResourceSelector'
}
