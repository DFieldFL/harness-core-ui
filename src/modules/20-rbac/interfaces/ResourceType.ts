export enum ResourceType {
  PROJECT = 'PROJECT',
  SECRET = 'SECRET',
  ORGANIZATION = 'ORGANIZATION',
  SECRET_MANAGER = 'SECRET_MANAGER'
}
