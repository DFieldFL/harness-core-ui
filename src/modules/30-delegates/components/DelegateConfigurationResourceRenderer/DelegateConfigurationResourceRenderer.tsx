import React, { useMemo } from 'react'
import get from 'lodash-es/get'
import { Button } from '@wings-software/uicore'
import type { RbacResourceRendererProps } from '@rbac/factories/RbacFactory'
import { useListDelegateProfilesNg } from 'services/cd-ng'
import type { DelegateProfile } from 'services/portal'
import Table from '@common/components/Table/Table'
import { PageSpinner } from '@common/components'

type CellType = { row: { original: DelegateProfile } }

const DelegateConfigurationResourceRenderer: React.FC<RbacResourceRendererProps> = ({
  identifiers,
  resourceScope,
  resourceType,
  onResourceSelectionChange
}) => {
  const { accountIdentifier, orgIdentifier, projectIdentifier } = resourceScope
  const queryParams = useMemo(
    () => ({
      accountId: accountIdentifier,
      orgId: orgIdentifier,
      projectId: projectIdentifier,
      module
    }),
    [accountIdentifier, orgIdentifier, projectIdentifier]
  )
  const { data, loading } = useListDelegateProfilesNg({ queryParams })

  const configurations: DelegateProfile[] = get(data, 'resource.response', [])

  const filteredConfigurations = configurations.filter(config => identifiers.includes(config.uuid))

  return filteredConfigurations && !loading ? (
    <Table
      columns={[
        {
          Header: '',
          id: 'name',
          accessor: 'name',
          width: '95%',
          Cell: (element: CellType) => <div>{element.row.original.name}</div>,
          disableSortBy: true
        },
        {
          id: 'removeBtn',
          width: '5%',
          disableSortBy: true,
          // eslint-disable-next-line react/display-name
          Cell: (element: CellType) => {
            return (
              <Button
                icon="trash"
                minimal
                onClick={() => {
                  onResourceSelectionChange(resourceType, false, [element.row.original.uuid])
                }}
              />
            )
          }
        }
      ]}
      data={filteredConfigurations}
      hideHeaders={true}
    />
  ) : (
    <PageSpinner />
  )
}

export default DelegateConfigurationResourceRenderer
