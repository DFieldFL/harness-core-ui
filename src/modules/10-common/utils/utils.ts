import type { IconName } from '@wings-software/uicore'
import { Scope } from '@common/interfaces/SecretsInterface'
import { ModuleName } from 'framework/types/ModuleName'

interface SetPageNumberProps {
  setPage: (value: React.SetStateAction<number>) => void
  pageItemsCount?: number
  page: number
}

export const getModuleIcon = (module: ModuleName): IconName => {
  switch (module) {
    case ModuleName.CD:
      return 'cd-main'
    case ModuleName.CV:
      return 'cv-main'
    case ModuleName.CI:
      return 'ci-main'
    case ModuleName.CE:
      return 'ce-main'
    case ModuleName.CF:
      return 'cf-main'
  }
  return 'nav-project'
}

export const getReference = (scope?: Scope, identifier?: string): string | undefined => {
  switch (scope) {
    case Scope.PROJECT:
      return identifier
    case Scope.ORG:
      return `org.${identifier}`
    case Scope.ACCOUNT:
      return `account.${identifier}`
  }
}

export const setPageNumber = ({ setPage, pageItemsCount, page }: SetPageNumberProps): void => {
  if (pageItemsCount === 0 && page > 0) {
    setPage(page - 1)
  }
}
