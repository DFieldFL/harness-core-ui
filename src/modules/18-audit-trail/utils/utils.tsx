import React from 'react'
import { Menu } from '@blueprintjs/core'
import type { ItemRenderer } from '@blueprintjs/select'
import { Layout, Text, Avatar, Color, MultiSelectOption } from '@wings-software/uicore'
import { isEmail } from '@common/utils/Validation'

export interface UserItem extends MultiSelectOption {
  email?: string
}

export const UserItemRenderer: ItemRenderer<UserItem> = (item, { handleClick }) => (
  <Menu.Item
    key={item.value.toString()}
    text={
      <Layout.Horizontal spacing="small" flex={{ alignItems: 'center', justifyContent: 'flex-start' }}>
        <Avatar name={item.label} email={item.email || item.value.toString()} size="small" hoverCard={false} />
        <Layout.Vertical padding={{ left: 'small' }}>
          <Text color={Color.BLACK}>{item.label}</Text>
          <Text color={Color.GREY_700}>{item.email || item.value}</Text>
        </Layout.Vertical>
      </Layout.Horizontal>
    }
    onClick={handleClick}
  />
)

export const UserTagRenderer = (item: UserItem, validate = false): React.ReactNode => (
  <Layout.Horizontal key={item.value.toString()} flex spacing="small">
    <Avatar name={item.label} email={item.value.toString()} size="xsmall" hoverCard={false} />
    <Text color={validate && !isEmail(item.value.toString().toLowerCase()) ? Color.RED_500 : Color.BLACK}>
      {item.label}
    </Text>
  </Layout.Horizontal>
)
