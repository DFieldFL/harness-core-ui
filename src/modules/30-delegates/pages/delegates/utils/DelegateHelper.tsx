import { useStrings } from 'framework/strings'
import { DelegateTypes } from '@delegates/constants'

export const GetDelegateTitleTextByType = (type: string): string => {
  const { getString } = useStrings()

  switch (type) {
    case DelegateTypes.KUBERNETES_CLUSTER:
      return getString('kubernetesText')
    default:
      /* istanbul ignore next */
      return ''
  }
}
