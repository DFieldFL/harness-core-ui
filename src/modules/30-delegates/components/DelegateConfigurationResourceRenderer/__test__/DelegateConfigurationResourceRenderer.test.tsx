import React from 'react'
import { render } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import { ResourceType } from '@rbac/interfaces/ResourceType'
import DelegateConfigurationResourceRenderer from '../DelegateConfigurationResourceRenderer'

const onChangeFn = jest.fn()

jest.mock('services/cd-ng', () => ({
  useListDelegateProfilesNg: jest
    .fn()
    .mockImplementation(() => ({ data: { content: [] }, refetch: jest.fn(), error: null, loading: false }))
}))

const resourceScope = {
  accountIdentifier: 'accountId',
  projectIdentifier: '',
  orgIdentifier: ''
}
const params = {
  identifiers: ['asd', 'esd'],
  resourceScope: resourceScope,
  resourceType: ResourceType.DELEGATECONFIGURATION,
  onResourceSelectionChange: onChangeFn
}

describe('Create DelegateConfigurationResourceRenderer', () => {
  test('render data', () => {
    const { container } = render(
      <TestWrapper>
        <DelegateConfigurationResourceRenderer {...params} />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })
})
