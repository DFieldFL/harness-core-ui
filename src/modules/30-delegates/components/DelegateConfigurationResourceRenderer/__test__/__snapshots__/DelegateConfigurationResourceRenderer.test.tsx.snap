// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create DelegateConfigurationResourceRenderer render data 1`] = `
<div>
  <div
    class="table"
  >
    <div
      class="body"
    />
  </div>
</div>
`;
